### ISSUE TITLE
can not use in jupyter notebook

### ISSUE THREAD

**Issue by hsz1273327** [tobgu/pyrthon#1]
there's a problem when woking with jupyter notebook

**Comment by tobgu** [comment]
OK, could you elaborate? Error messages? Misbehaviors?

**Comment by hsz1273327** [comment]
[1.PNG.zip](https://github.com/tobgu/pyrthon/files/177350/1.PNG.zip)
![1](https://cloud.githubusercontent.com/assets/7850148/13838108/46e70642-ec4c-11e5-89f1-3e1f96573a40.PNG)

**Comment by tobgu** [comment]
And what exactly is the problem? What is the output if you type [1, 2, 3] after enabling it?

**Comment by hsz1273327** [comment]
er, just like below,I can only use the cell with "import pyrthon.console " ,this situation is like using "input()"
![2016-03-17 3 39 37](https://cloud.githubusercontent.com/assets/7850148/13839554/d19d9d8c-ec56-11e5-9088-1674ad036c53.png)

**Comment by tobgu** [comment]
OK, I'm not familiar with the jupyter notebook at all. Are you saying that you would like the change to take effect across cells (and it does not currently)? Is that how imports work for example, eg. importing in one cell will make the imported thing available in all cells?

**Comment by hsz1273327** [comment]
yes, i think that is the correct behavior. 

来自 魅蓝 note 2

-------- 原始邮件 --------
发件人：Tobias Gustafsson notifications@github.com
时间：周四 3月17日 下午 03:50
收件人：tobgu/pyrthon pyrthon@noreply.github.com
抄送：HUANG SIZHE hsz1273327@gmail.com
主题：Re: [pyrthon] can not use in jupyter notebook (#1)

**Comment by tobgu** [comment]
OK, I think I understand your problem. I don't have any simple solution to it though. Anyone with more experience from Jupyter/IPython is most welcome to help out!

**Comment by yardsale8** [comment]
Here is a quick fix.  Ipython comes with a hook for ast_transformations.  You can use `prython` in the ipython notebook/console/qtconsole by running the following lines.

``` python
from pyrthon._transformer import PyrsistentTransformer
pyr_transform = PyrsistentTransformer()
ip = get_ipython()
ip.ast_transformers.append(pyr_transform)
```
After running these cell(s) you get automatic conversion to `pyrsistent` data structures.  Here is the input and output for the next cell
``` python
In [2]:      type([])
Out [2]:   pvectorc.PVector
```
It should be pretty easy to set up a `%load_Ext` command to run the set up lines, I will look into it.

**Comment by yardsale8** [comment]
I have submitted a pull request #2 that implements the ipython loading of `pyrthon` using `%load_ext pyrthon`.

**Comment by tobgu** [comment]
Fixed and released in v0.1.2. Thanks @yardsale8!
